projects/packages/account-protection/src/class-validation-service.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Class used to define Validation Service.
+ *
+ * @package automattic/jetpack-account-protection
+ */
+
+namespace Automattic\Jetpack\Account_Protection;
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+
+/**
+ * Class Validation_Service
+ */
+class Validation_Service {
+	/**
+	 * Check if the password is in the list of compromised/common passwords.
+	 *
+	 * @param string $password The password to check.
+	 * @return bool|\WP_Error True if the password is in the list of compromised/common passwords, false otherwise.
+	 */
+	public function check_weak_passwords( string $password ) {
+		$api_url = '/jetpack-protect-weak-password';
+
+		$is_connected = ( new Connection_Manager() )->is_connected();
+		if ( ! $is_connected ) {
+			return new \WP_Error( 'site_not_connected' );
+		}
+
+		$hashed_password = sha1( $password );
+		$password_prefix = substr( $hashed_password, 0, 5 );
+
+		$response = Client::wpcom_json_api_request_as_blog(
+			$api_url . '/' . $password_prefix,
+			'2',
+			array( 'method' => 'GET' ),
+			null,
+			'wpcom'
+		);
+
+		$response_code = wp_remote_retrieve_response_code( $response );
+
+		if ( is_wp_error( $response ) || 200 !== $response_code || empty( $response['body'] ) ) {
+			return false;
+			// TODO: Return false or log error?
```

Comment:
we should probably be returning true in this case actually, because otherwise if our API goes down or something we would be blocking all user logins? 🤔

Reply:
This particular method returns `true` if the password is compromised or common and `false` otherwise, so when the endpoint callback errors out, is non-200, or is empty the user should be able to login every time. 

My TODO comment there is for whether we want to report this is some way so the user is aware that their login is not actually secured. At least on our end, I suspect that if this happens often we'll see these entries in logstash.